apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts

```diff
@@ -1,58 +1,311 @@
-    // mock the query method that is used
-    query: jest.fn(() => Promise.resolve({}))
-  };
-
-  const mockedTransactionService = {
-    // mock the query method that is used
-    query: jest.fn(() => Promise.resolve({}))
-  };
+  let posDepositService: DeepMocked<PosDepositService>;
+  let paymentService: DeepMocked<PaymentService>;
+  let logger: DeepMocked<AppLogger>;
+  let payments: PaymentEntity[];
+  let deposits: POSDepositEntity[];
+  let paymentExceptions: PaymentEntity[];
+  let depositExceptions: POSDepositEntity[];
+  let matchedRoundOne: { payment: PaymentEntity; deposit: POSDepositEntity }[];
+  let matchedRoundTwo: { payment: PaymentEntity; deposit: POSDepositEntity }[];
+  let matches: { payment: PaymentEntity; deposit: POSDepositEntity }[];
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [
-        POSReconciliationService,
-        PosDepositService,
-        LocationService,
-        Logger,
-        {
-          provide: getRepositoryToken(POSDepositEntity),
-          useValue: mockedRepo
-        },
-        {
-          provide: getRepositoryToken(LocationEntity),
-          useValue: mockedRepo
-        },
-        {
-          provide: TransactionService,
-          useValue: mockedTransactionService
-        },
-        {
-          provide: PaymentService,
-          useValue: mockedPaymentService
-        }
-      ]
-    }).compile();
+      providers: [POSReconciliationService]
+    })
+      .useMocker(createMock)
+      .compile();
+
+    const mockPOSData = new MockData(PaymentMethodClassification.POS);
+
+    payments = mockPOSData.paymentsMock;
+    /**
+     * Set the transaction time to be 20 minutes after the transaction date for one payment
+     */
+    payments[0].transaction.transaction_time = format(
+      getTime(
+        parse(
+          payments[0].transaction.transaction_date +
+            ' ' +
+            payments[0].transaction.transaction_time,
+          'yyyy-MM-dd HH:mm:ss',
+          new Date()
+        )
+      ) +
+        1000 * 60 * 20,
+      'HH:mm:ss'
+    );
+    deposits = mockPOSData.depositsMock as POSDepositEntity[];
 
     service = module.get<POSReconciliationService>(POSReconciliationService);
+    posDepositService = module.get(PosDepositService);
+    paymentService = module.get(PaymentService);
+    logger = module.get(Logger);
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
+    expect(posDepositService).toBeDefined();
+    expect(paymentService).toBeDefined();
+    expect(logger).toBeDefined();
+  });
+
+  it('should verify the test data transaction_time', () => {
+    expect(
+      differenceInMinutes(
+        parse(payments[0].transaction.transaction_time, 'HH:mm:ss', new Date()),
+        parse(deposits[0].transaction_time, 'HH:mm:ss', new Date())
+      )
+    ).toBeGreaterThanOrEqual(20);
```

Comment:
What exactly are we testing here? Just that the 20 minute addition worked? It feels like we're just testing date-fns `differenceInMinutes`